aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -185,12 +185,20 @@ impl SnapshotType {
 
 fn get_context_data<'t, 'b>(
     context: &'t mut SafeNativeContext<'_, 'b, '_, '_>,
-) -> (&'b dyn DelayedFieldResolver, RefMut<'t, DelayedFieldData>) {
+) -> Option<(&'b dyn DelayedFieldResolver, RefMut<'t, DelayedFieldData>)> {
     let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
-    (
-        aggregator_context.resolver,
-        aggregator_context.delayed_field_data.borrow_mut(),
-    )
+    if aggregator_context
+        .delayed_field_resolver
+        .is_delayed_field_optimization_capable()
+        && context.aggregator_v2_delayed_fields_enabled()
```

Comment:
What's the difference between `is_delayed_field_optimization_capable` and `aggregator_v2_delayed_fields_enabled` flags?

Reply:
is_delayed_field_optimization_capable means whether resolver we are using is capable of using that feature.

aggregator_v2_delayed_fields_enabled means whether a feature has been enabled.

we need both capable resolver (i.e. either one from BlockExecutor) and feature flag enabled.

for example, executing view functions is outside of BlockExecutor, and even after flag is enabled, resolver will not be capable of handling it.